platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/implementing_custom_tasks.adoc

```diff
@@ -429,16 +433,18 @@ public abstract class GenerateReportTask : DefaultTask() {
 [source,groovy]
 ----
 import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.OutputFile
 
-public abstract class GenerateReportTask extends DefaultTask {
+abstract class GenerateReportTask extends DefaultTask {
```

Comment:
In Java, `package-private` is the default visibility, so users won't be able to use them, afaik.

Reply:
This is change for a Groovy example file, so public is default similarly as for Kotlin